### PR TITLE
Increase redisvl test coverage

### PR DESCRIPTION
**Pull request by bsbodden**
Upped the coverage to 81% and fix a minor bug in the utils

### CODE REVIEW

File: redisvl/redis/utils.py [hunk 1]
```diff
@@ -20,12 +20,12 @@ def convert_bytes(data: Any) -> Any:
             return data.decode("utf-8")
         except:
             return data
-    if isinstance(data, dict):
```
Comment (tylerhutcherson) [inline redisvl/redis/utils.py:23]: just curious why `elif` given the return path?

File: tests/integration/test_llmcache.py [hunk 2]
```diff
@@ -130,3 +160,26 @@ def test_multiple_items(cache, vectorizer):
         print(check_result, flush=True)
         assert check_result[0]["response"] == expected_response
         assert "metadata" not in check_result[0]
+
+# Test retrieving underlying SearchIndex for the cache.
+def test_get_index(cache):
+    assert type(cache.index) is SearchIndex
```
Comment (tylerhutcherson) [inline tests/integration/test_llmcache.py]: `assert isinstance(cache.index, SearchIndex)` is the preferred python check for this!

File: tests/unit/test_async_search_index.py [hunk 3]
```diff
@@ -137,3 +138,36 @@ async def test_no_id_field(async_client, async_index):
...
+@pytest.mark.asyncio
+async def test_check_index_exists_before_delete(async_client, async_index):
+    async_index.set_client(async_client)
+    await async_index.create(overwrite=True, drop=True)
+    await async_index.delete(drop=True)
+    with pytest.raises(ValueError):
+        await async_index.delete()
+
+@pytest.mark.asyncio
+async def test_check_index_exists_before_search(async_client, async_index):
```
Comment (tylerhutcherson) [inline tests/unit/test_async_search_index.py:152]: awesome thanks for adding these checks!

File: tests/unit/test_utils.py
Comment (tylerhutcherson) [inline tests/unit/test_utils.py:1]: huge! thanks!

**Review by tylerhutcherson — COMMENTED** [review]
Looks solid! I left a few comments. We'll get this merged today.

File: redisvl/redis/utils.py (hunk 1 above)
Comment (bsbodden) [reply redisvl/redis/utils.py:23]: Changed it back to short-circuit ifs

File: tests/integration/test_llmcache.py (hunk 2 above)
Comment (bsbodden) [reply tests/integration/test_llmcache.py]: Changed!

**Review by tylerhutcherson — APPROVED** [review]

**Pull request merged**